spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -229,12 +225,7 @@ public DeploymentState getState() {
 			Integer exit = getProcessExitValue(process);
 			// TODO: consider using exit code mapper concept from batch
 			if (exit != null) {
-				if (exit == 0) {
```

Comment:
Thinking out loud here, without having had a chance to test this:
given that the key in the map is the app name, wouldn't this prevent a redeployment of an app?

Shouldn't we rather report a `== 0 `exited process as `unknown` (_i.e._ as if it never exi(s)ted) and a `!= 0` exited process as `failed`?